cmd/kubeadm/app/cmd/init.go

```diff
@@ -374,3 +387,37 @@ func createClientsetAndOptionallyWaitForReady(cfg *kubeadmapi.MasterConfiguratio
 	}
 	return client, nil
 }
+
+func getDirectoriesToUse(dryRun bool, defaultPkiDir string) (string, string, string, error) {
```

Comment:
I hope we can defer that until after code freeze. Agree that it would be nice to refactor this once we have the full picture afterwards.